lib/api/CInferenceModelDefinition.cc

```diff
@@ -408,6 +533,62 @@ void CInferenceModelDefinition::dependentVariableColumnIndex(std::size_t depende
     m_DependentVariableColumnIndex = dependentVariableColumnIndex;
 }
 
+CInferenceModelDefinition::TSizeInfoUPtr CInferenceModelDefinition::sizeInfo() const {
+    return std::make_unique<CSizeInfo>(*this);
+}
+
+CInferenceModelDefinition::CSizeInfo::CSizeInfo(const CInferenceModelDefinition& definition)
+    : m_Definition{definition} {
+}
+
+std::string CInferenceModelDefinition::CSizeInfo::jsonString() {
+    std::ostringstream stream;
+    {
+        core::CJsonOutputStreamWrapper wrapper{stream};
+        CSerializableToJson::TRapidJsonWriter writer{wrapper};
+        rapidjson::Value doc = writer.makeObject();
+        this->addToDocument(doc, writer);
+        writer.write(doc);
+        stream.flush();
+    }
+    // string writer puts the json object in an array, so we strip the external brackets
+    std::string jsonStr{stream.str()};
+    std::string resultString(jsonStr, 1, jsonStr.size() - 2);
+    return resultString;
+}
+
+void CInferenceModelDefinition::CSizeInfo::addToDocument(rapidjson::Value& parentObject,
+                                                         TRapidJsonWriter& writer) const {
+    // parse preprocessing
+    TEncodingSizeUPtrVec encodingSizeItems;
+    encodingSizeItems.reserve(m_Definition.preprocessors().size());
+    for (const auto& preprocessor : m_Definition.preprocessors()) {
+        encodingSizeItems.push_back(preprocessor->sizeInfo());
+    }
```

Comment:
Is there any reason to copy these into a vector? It strikes me you could change:
```
for (const auto& encoding : encodingSizeItems) {
```
to 
```
for (const auto& preprocessor : m_Definition.preprocessors()) {
    auto encodingSizeInfo = preprocessor->sizeInfo();
```

Reply:
You are right, it further simplifies things.